compiler/executable/match_/planner/plan.rs

```diff
@@ -810,6 +758,28 @@ impl<'a> ConjunctionPlanBuilder<'a> {
     }
 }
 
+struct DrainSorted<'a,T: Ord> {
+    heap: &'a mut BinaryHeap<T>
+}
+
+fn drain_sorted<T: Ord>(heap: &mut BinaryHeap<T>) -> impl Iterator<Item=T> + '_ {
+    DrainSorted { heap }
+}
+
+impl<'a, T: Ord> Iterator for DrainSorted<'a, T> {
+    type Item = T;
+
+    fn next(&mut self) -> Option<Self::Item> {
+        self.heap.pop()
+    }
+}
+
+impl<'a, T: Ord> Drop for DrainSorted<'a, T> {
+    fn drop(&mut self) {
+        self.heap.clear();
+    }
+}
+
```

Comment:
[Nightly](https://doc.rust-lang.org/std/collections/binary_heap/struct.DrainSorted.html) but i wanted it because it makes things nice and readable.